rotkehlchen/balances/historical.py

```diff
@@ -99,63 +265,230 @@ def get_assets_amounts(
+        missing_price_points: list[tuple[str, Timestamp]] = []
+        for day_ts, asset_balances in daily_balances.items():
+            assets_to_price = [
+                asset for asset, balance in asset_balances.items()
+                if balance != ZERO
+            ]
+            if len(assets_to_price) == 0:
+                continue
+
+            prices, missing = self._get_prices(
+                timestamp=day_ts,
+                assets=assets_to_price,
+                main_currency=main_currency,
+            )
+            missing_price_points.extend(missing)
+
+            day_total = sum(
+                balance * prices[asset]
+                for asset, balance in asset_balances.items()
+                if asset in prices and balance != ZERO
+            )
+            net_worth_per_day[day_ts] = FVal(day_total)
+
+        return net_worth_per_day, missing_price_points, negative_balance_event
 
     def _get_events_and_currency(
             self,
             from_ts: Timestamp | None = None,
             to_ts: Timestamp | None = None,
             assets: tuple[Asset, ...] | None = None,
-    ) -> tuple[list['HistoryBaseEntry'], Asset]:
-        """Helper method to get historical events and main currency from DB"""
-        db_history_events = DBHistoryEvents(database=self.db)
-        with self.db.conn.read_ctx() as cursor:
-            events = db_history_events.get_history_events(
-                cursor=cursor,
-                filter_query=HistoryEventFilterQuery.make(
-                    from_ts=from_ts,
-                    to_ts=to_ts,
-                    assets=assets,
-                ),
-                has_premium=True,
+    ) -> tuple[list[MinimalHistoryEvent | MinimalTrade], Asset]:
+        """Helper method to get events(including trades) and main currency from DB.
+        Kraken trades are ignored because they are duplicated in the history events table.
+
+        Using two cursors to parallel scan trades and history_events tables, merging them
+        chronologically. This avoids loading full tables into memory while maintaining
+        event ordering. Each cursor fetches one table, allowing streaming iteration.
+
+        TODO: Refactor this once trades are migrated to history events.
+        """
+        events: list[MinimalHistoryEvent | MinimalTrade] = []
+        with self.db.conn.read_ctx() as cursor1, self.db.conn.read_ctx() as cursor2:
+            main_currency = self.db.get_setting(cursor=cursor1, name='main_currency')
+            trade_filters: list[DBFilter] = [
+                DBNotEqualFilter(and_op=True, column='location', value='B'),
+                DBTimestampFilter(and_op=True, from_ts=from_ts, to_ts=to_ts),
+            ]
+            event_filters: list[DBFilter] = [DBTimestampFilter(
+                and_op=True,
+                to_ts=to_ts,
+                from_ts=from_ts,
+                scaling_factor=FVal(1000),
+            )]
+
+            if assets is not None:
```

Comment:
Why is assets optional in this function? Add a comment explaining what you do in this codeblock